File: aat-lib/src/main/java/ch/bailu/aat_lib/gpx/information/GpxInformationCache.java
package ch.bailu.aat_lib.gpx.information;

import ch.bailu.aat_lib.dispatcher.TargetInterface;

public class GpxInformationCache {
    public GpxInformation info;
    public int infoID;


    public GpxInformationCache() {
        reset();
    }


    public void set(int _infoID, GpxInformation _info) {
        info = _info;
        infoID = _infoID;
    }


    public void reset() {
        set(InfoID.UNSPECIFIED, GpxInformation.NULL);
    }


    public void letUpdate(TargetInterface obj) {
        obj.onContentUpdated(infoID, info);
    }
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/gpx/information/GpxInformationCache.kt
package ch.bailu.aat_lib.gpx.information

import ch.bailu.aat_lib.dispatcher.TargetInterface

class GpxInformationCache {
    var info = GpxInformation.NULL
    var infoID = InfoID.UNSPECIFIED

    init {
        reset()
    }

    fun set(iid: Int, info: GpxInformation) {
        this.info = info
        infoID = iid
    }

    fun reset() {
        set(InfoID.UNSPECIFIED, GpxInformation.NULL)
    }

    fun letUpdate(obj: TargetInterface) {
        obj.onContentUpdated(infoID, info)
    }
}
